app/src/components/SoldArtworkList.tsx

```diff
@@ -42,17 +42,18 @@ class SoldArtworkList extends React.Component<ContractProps, SoldArtworkListStat
 
     await governance.getPastEvents(
       'RecordARR', options).then(function (events: any) {
-      eventsFound = events.length;
       for (let i = 0; i < events.length; i++) {
-        const soldInfo: SoldInformation = {
-          tokenId: events[i].returnValues.tokenId,
-          price: events[i].returnValues.price,
-          newOwner: events[i].returnValues.to,
-        };
-        soldInfoArray.push(soldInfo);
+        if (events[i].returnValues.from === currentAccount) {
+          eventsFound += 1;
```

Comment:
do you not need an eventsFound = 0 somewhere to instantiate this?

Reply:
```
const SoldArtworkList: React.FC<ContractProps> = ({contracts, accounts}) => {
  const [balance, setBalance] = React.useState<number>(-1);
  const [soldList, setSoldList] = React.useState<SoldInformation[]>([]);

  const loadSales: VoidFunction = () => {
    const governance = contracts.Governance;
    const currentAccount = accounts[0];
    const filter = { filter: { from: currentAccount } };
    const options = { filter, fromBlock: 0 };

    governance.getPastEvents('RecordARR', options).then((events: any[]) => {
      setSoldList(events.map((event) => {
          return {
            tokenId: event.returnValues.tokenId,
            price: event.returnValues.price,
            newOwner: event.returnValues.to,
          };
        }
      ));

      setBalance(events.length);
    }).catch(console.log);
  }

  React.useEffect(loadSales, [balance, soldList]);

  if (balance === -1) {
    return (
      <h1>No sales to display</h1>
    );
  }

  const listItems = soldList.map((sold, index) => {
    return <SoldArtworkItem
      contracts={contracts}
      accounts={accounts}
      soldFor={sold.price}
      soldTo={sold.newOwner}
      tokenId={sold.tokenId}
      key={sold.tokenId * 3 + index * 5}
    />
  });

  return (
    <CardColumns>{listItems}</CardColumns>
  );

};

```

Reply:
Stumbled into this on my branch ^ This was my fix

Reply:
Neil no I don’t, that’s done outside of my loops, Dan what are you attempting to fix there?

Reply:
Oh the key values? That should be a super easy fix. Dan are you sure that with that code the filter correctly only returns sales that your account has made?

Reply:
Was also talking about the way you use eventsFound that this gets rid of